aiohttp/client_exceptions.py

```diff
@@ -238,17 +239,34 @@ class InvalidURL(ClientError, ValueError):
 
     # Derive from ValueError for backward compatibility
 
-    def __init__(self, url: Any) -> None:
+    def __init__(self, url: StrOrURL, description: Optional[str] = None) -> None:
         # The type of url is not yarl.URL because the exception can be raised
         # on URL(url) call
-        super().__init__(url)
+        if description:
+            super().__init__(url, description)
+        else:
+            super().__init__(url)
+
+    @property
+    def url(self) -> StrOrURL:
+        return self.args[0]  # type: ignore[no-any-return]
 
     @property
-    def url(self) -> Any:
-        return self.args[0]
+    def description(self) -> Optional[str]:
+        try:
+            return self.args[1]  # type: ignore[no-any-return]
+        except KeyError:
+            return None
 
     def __repr__(self) -> str:
-        return f"<{self.__class__.__name__} {self.url}>"
+        return f"<{self.__class__.__name__} {self}>"
+
+    def __str__(self) -> str:
+        return " - ".join(map(str, self.args))
```

Comment:
I'd probably make this colon- or period-separated instead.

Reply:
Can You post an example? I'm not sure what You mean

Reply:
`" : "`

I don't really mind regardless.

Reply:
I was thinking maybe
```suggestion
        return ". ".join(map(str, self.args))
```

Reply:
`" : "` looks better than dot for me but `" - "` seems to be more general (in case of more args). I don't really mind too so just let me know which version you will accept so I will keep current one or change

Reply:
With the suggestions to actually save the args into private attributes, I'd drop the loop here and use an f-string for clarity.

Reply:
updated